### PR TITLE
Added the bill and updated relevant functions

### PR DESCRIPTION
**Pull request by meowbot24**
Summary of changes:
- Created a bill to keep track of items purchased
- Modified void item functions so bill will be updated accordingly
- Bill modified to consolidate same items, rather than store each instance of each item separately

### CODE REVIEW

File: cash-register.js [hunk 1]
```diff
@@ -6,42 +6,131 @@
...
+
+- input for staff discount
+- ability to add new items
+- void last transaction
+- void any item + quantity
+
+*/
 
 var cashRegister = {
 	total: 0,
```
Comment (scottybarr) [inline cash-register.js]: Use 2 spaces instead of tabs. (hint lookup: _eslint_)

File: cash-register.js [hunk 2]
```diff
@@ -6,42 +6,131 @@
...
 	lastItemCost: 0,
+	lastItemName: "",
```
Comment (scottybarr) [inline cash-register.js]: Use single quotes ' vs " (hint _eslint_)

File: cash-register.js [hunk 3]
```diff
@@ -6,42 +6,131 @@
...
 
+		// keep track of what the last item and quantity was for bill
+		this.lastItemName = itemName;
+		this.lastItemQuantity = quantity;
+
 		// looping through itemList to match name to find price & add to total
 		for (var x in itemList) {
 			if (itemName === itemList[x]["name"]) {
-				console.log(itemList[x]["name"]);
+				console.log(itemList[x]["name"] + ", " + quantity);
```
Comment (scottybarr) [inline cash-register.js]: For the future: If you put code in production - remove `console.log` :)

File: cash-register.js [hunk 4]
```diff
@@ -6,42 +6,131 @@
...
+			};
+		};
+
+		// add item to bill
+
+		// check if item exists on bill
+
+		var billItemCheck = false;
+
+		for (var m in this.bill) {
```
Comment (scottybarr) [inline cash-register.js]: `var m` - Use descriptive var names :)

File: cash-register.js [hunk 5]
```diff
@@ -6,42 +6,131 @@
...
+			if (itemName === this.bill[m]["name"]) {
+				// console.log(itemName + " exists on the bill.");
+				billItemCheck = true;
+			} else {					
+				// console.log(itemName + " does not exist on the bill");
```
Comment (scottybarr) [inline cash-register.js]: @meowbot24 You don't need the else :)

File: cash-register.js [hunk 6]
```diff
@@ -6,42 +6,131 @@
...
 
 	voidLastItem: function() {
 		this.total -= this.lastItemCost;
-		console.log("The last item has been voided, totalling: " + this.lastItemCost);
+
+		// delete last scanned item & quantity from the bill
+
+		for (var x in this.bill) {
+			if (this.lastItemName === this.bill[x]["name"]) {
+				if (this.bill[x]["quantity"] > this.lastItemQuantity) {
```
Comment (scottybarr) [inline cash-register.js]: @meowbot24 Add a comment for this line to indicate why this would occur.

File: cash-register.js [hunk 7]
```diff
@@ -88,20 +178,27 @@ addNewItem("popcorn",5);
 // 	console.log(itemList[x]["price"]);
 // };
 
+
 cashRegister.add("milk",6); // add 6 milk to total
 
+cashRegister.add("kitty food",2); // add 2 kitty food to total
```
Comment (scottybarr) [inline cash-register.js]: @meowbot24 space between the comma and the 2. :) (hint _eslint_)

File: cash-register.js (hunk 1 above)
Comment (meowbot24) [inline cash-register.js]: Done!

File: cash-register.js (hunk 2 above)
Comment (meowbot24) [inline cash-register.js]: Done!

File: cash-register.js (hunk 3 above)
Comment (meowbot24) [inline cash-register.js]: Will do! I'm using console.log to generate 'outputs' from my code, I will definitely have to rewrite to remedy this! Haha

File: cash-register.js (hunk 4 above)
Comment (meowbot24) [inline cash-register.js]: Done!

File: cash-register.js (hunk 5 above)
Comment (meowbot24) [inline cash-register.js]: Removed!

File: cash-register.js (hunk 6 above)
Comment (meowbot24) [inline cash-register.js]: Done!

File: cash-register.js (hunk 7 above)
Comment (meowbot24) [inline cash-register.js]: Sorry, made a mistake! I added the spaces in my local file, but the file I committed and pushed doesn't have the spaces. Will be up in my next commit!

File: cash-register.js [hunk 8]
```diff
@@ -6,102 +6,199 @@
```
Comment (scottybarr) [inline cash-register.js:97]: @meowbot24 I think this file needs deleted too! ;)

File: cash-register/index.html [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+</head>
+<body>
+	
+	<script type="text/javascript" src="js/Item.js"></script>
+	<script type="text/javascript" src="js/ItemList.js"></script>
+	<script type="text/javascript" src="js/CashRegister.js"></script>
+	<script type="text/javascript" src="js/app.js"></script>
+
+</body>
+</html>
```
Comment (scottybarr) [inline cash-register/index.html]: @meowbot24 Style - Add a newline at the end of files. The little 🚫⏎ icons here suggests we should add a newline.

File: cash-register/js/CashRegister.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+      }
+    }
+
+    // add item to bill
+
+    // check if item exists on bill
+
+    var billItemCheck = false;
+
+    for (var billItem in this.bill) {
```
Comment (scottybarr) [inline cash-register/js/CashRegister.js:30]: Okay, today's lesson - Leveraging other libraries (aka `Don't reinvent the wheel!`).
There's a very helpful utility library called `lodash` (https://lodash.com) which provides very useful functions across strings, arrays and objects. if you include that script in your HTML file then you could call the `_.includes()` function (defined at https://lodash.com/docs#includes) which would provide the same functionality as this `billItemCheck` loop for you. :)

Comment (scottybarr) [inline cash-register/js/CashRegister.js:30]: @meowbot24 Forgot to tag you in comment above.

File: cash-register/js/CashRegister.js [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        billItemCheck = true;
+      }
+    }
+
+
+    // add item or quantity to bill depending on if item is already on bill
+    
+    if (billItemCheck) {
+      // add quantity
+      for (var billItem in this.bill) {
```
Comment (scottybarr) [inline cash-register/js/CashRegister.js:41]: @meowbot24 Bonus points if you can use JavaScript's `reduce` function here. :)

Comment (scottybarr) [inline cash-register/js/CashRegister.js:41]: https://lodash.com/docs#reduce

File: cash-register/js/Item.js [hunk 12]
```diff
@@ -0,0 +1,5 @@
+// creating object class for Items
+function Item(name, price) {
+  this.name = name;
+  this.price = price;
+}
```
Comment (scottybarr) [inline cash-register/js/Item.js]: @meowbot24 Style - newline at the end of the file.

File: cash-register/js/ItemList.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+// add new items to itemList
+
+var addNewItem = function(name, price) {
+  itemList[name] = new Item(name,price);
+};
+
+addNewItem('candy',4);
+addNewItem('popcorn',5);
+
+// // to view/log itemList
```
Comment (scottybarr) [inline cash-register/js/ItemList.js]: @meowbot24 Style - If you don't want code then just delete it :) Git has version control to go back in history if you want it back. :)

File: cash-register/js/ItemList.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
+// object containing all items available for sale
+var itemList = {
+  milk: new Item('milk', 2.00),
+    eggs: new Item('eggs', 3.00),
```
Comment (scottybarr) [inline cash-register/js/ItemList.js]: @meowbot24 Style - Keep these attributes at the same level of indentation (see how `milk` and `eggs` are indented differently?)

File: cash-register/js/app.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+// cashRegister.voidItem('kitty food',1); // void 1 kitty food 
+
+cashRegister.add('milk', 4); // add 4 milk to total
+
+cashRegister.voidLastItem(); // void 6 milk
+
+cashRegister.add('kitty food', 2); // add 2 kitty food to total
+
+// cashRegister.add('popcorn',4);
```
Comment (scottybarr) [inline cash-register/js/app.js]: @meowbot24 Delete unused code ;)

File: cash-register.js (hunk 8 above)
Comment (meowbot24) [inline cash-register.js:97]: Okay done! :) So as a best practice, should I only keep code that should be reviewed up on Git? I shouldn't leave any 'copies' for reference for myself? I guess that's what local files are for right? Haha

File: cash-register/index.html (hunk 9 above)
Comment (meowbot24) [inline cash-register/index.html]: Awesome good to know, done! :)

File: cash-register/js/Item.js (hunk 12 above)
Comment (meowbot24) [inline cash-register/js/Item.js]: Done!

File: cash-register/js/ItemList.js (hunk 13 above)
Comment (meowbot24) [inline cash-register/js/ItemList.js]: Makes total sense, done! :)

File: cash-register/js/ItemList.js (hunk 14 above)
Comment (meowbot24) [inline cash-register/js/ItemList.js]: Right, makes sense! Fixed :D

File: cash-register/js/app.js (hunk 15 above)
Comment (meowbot24) [inline cash-register/js/app.js]: Done! Cleaned up app.js a bit :)

File: cash-register.js (hunk 8 above)
Comment (scottybarr) [inline cash-register.js:97]: Yep, only put code that's relevant in git. Depends on the usefulness of the "copy" for yourself? If it's useful then keep it in git and keep it well documented (as it's basically a knowledge document at that point). If it's not providing use then get rid of it. :)

**Comment by scottybarr** [comment]
Reviewed! Awesome work!
I think we can merge this and start working on the bonus points as a separate PR! 😸  Mao Mao!

**Pull request merged**